<h1 align= 'center'>Project: The Dumb Predictor</h1>
<h5>Mason Manca </h5>
<h5>CPSC 323 01 </h5>
<h5>Dr.Morehead </h5>
<h5>8 September 2022 </h5>



In [1]:
# Import cell 
import time
import pandas as pd



In [2]:
# This cell holds all methods    
class Model:
    def __init__(self):
        pass
    
    def predict(self,df):
        predictionList = []
        for row in enumerate(df.iterrows()):
            predictionList.append(1)       
    
        return(predictionList)


In [3]:
# Cell for output of data and time calculation
model = Model()
df = pd.read_csv('heart.csv')

t0 = time.time() # Start of the clock to determine performance

predictions = model.predict(df)

timeOfModel = time.time() - t0 # Calculation of performance speed

# Prediction output: 
# Always going to be false because it is the "Dumb Predictor"
print(predictions[:50], "\n Shortened to 50 for conciseness")


[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] 
 Shortened to 50 for conciseness


##### Calculating Accuracy, F1 Score and Confusion Matrix

Since we can only have 'false positives' and 'true positives' since we are predicting a boolean value with a model that only returns 'true'. We are able to determine accuracy simply by counting the number of correct identifications (TP) and dividing by all inputs.

The rest of the values are False positives.

In [4]:
# This cell is for Analysis of accuracy, F1 score 
# Creating variables for analysis


TP = int(df['HeartDisease'].value_counts()[1]) # Predicted positive, are positive
FP = int(df['HeartDisease'].value_counts()[0]) # Predicted positive are negative
FN = 0 # Predicted negative are positive
TN = 0 # Predicted negative are negative

allVals = 918

# Accuracy 
accuracy = TP/allVals # number of inputs
print('Accuracy: ', round(accuracy,2))


# F1 score: F1 = 2 * (precision * recall) / (precision + recall)

precision = TP/(TP + FP) # in this case because of the model TP/(TP+FP)
recall = TP/(TP + FN)

F1_score = 2 * ((precision * recall) / (precision + recall))
print("F1 Score: ", round(F1_score,2))

print("Speed of Model: ", round(timeOfModel,2))

Accuracy:  0.55
F1 Score:  0.71
Speed of Model:  0.04


##### Confusion Matrix

|       | Predicted (0)  | Predicted (918)|
| ----------- | ----------- |----------- |
|Actual (410)| TN    (0)   | FP (410)    |
|Actual (508)| FN    (0)   | TP  (508)   |

Accuracy Analysis:
- 55% accurate over 918 cases

F1 Score Analysis:
- Harmonic mean of .71

Confusion Matrix Analysis:
- 410 False Positives
- 508 True Positives

Performance Speed:
- Approx. 0.03 sec over 918 instances

<h4>Conclusion: A dumb predictor</h4>
    